dcc-portal-ui/app/scripts/compounds/js/compounds.js

```diff
@@ -361,8 +361,8 @@ angular.module('icgc.compounds.services', ['icgc.genes.models'])
             _compoundEntity = _compoundEntityFactory(compound.plain());
             defer.resolve(_self);
           }, function(){
-            Notify.hide();
-            $state.go('404', {page: 'compound', id: compoundId});
+            Page.stopWork();
+            $state.go('404', {page: 'compound', id: compoundId, url: '/compound/:id'}, {location: false});
```

Comment:
For compounds page, yes. Because it is working differently then other entity pages. The resolve function returns a promise instead of the actual compound entity date. But yes, you are right I should add a check if the error is 404 or not.